lib/improver/nbhood/recursive_filter.py

```diff
@@ -0,0 +1,394 @@
+                Bi-1 = New value at gridpoint i-1
+
+        In the backwards direction:
+            Recursive filtering is calculated as:
+                Bi = ((1-alpha) * Ai) + (alpha * Bi+1)
+
+            Progressing from gridpoint i+1 to i:
+                Bi = new value at gridpoint i, Ai = Old value at gridpoint i
+                Bi+1 = New value at gridpoint i+1
+
+        Args:
+            cube (Iris.cube.Cube):
+                Cube containing the input data to which the recursive filter
+                will be applied.
+            alphas_x (Iris.cube.Cube):
+                Cube containing array of alpha values that will be used
+                when applying the recursive filter along the x-axis.
+            alphas_y (Iris.cube.Cube):
+                Cube containing array of alpha values that will be used
+                when applying the recursive filter along the y-axis.
+            iterations (integer):
+                The number of iterations of the recursive filter
+
+        Returns:
+            cube (Iris.cube.Cube):
+                Cube containing the smoothed field after the recursive filter
+                method has been applied to the input cube.
+        """
+
+        output = cube.data
+        for _ in range(iterations):
+            output = RecursiveFilter.recurse_forward_x(output,
+                                                       alphas_x.data)
+            output = RecursiveFilter.recurse_backwards_x(output,
+                                                         alphas_x.data)
+            output = RecursiveFilter.recurse_forward_y(output,
+                                                       alphas_y.data)
+            output = RecursiveFilter.recurse_backwards_y(output,
+                                                         alphas_y.data)
+            cube.data = output
+        return cube
+
+    def set_alphas(self, cube, alpha, alphas_cube):
+        """
+        Set up the alpha parameter.
+
+        Args:
+            cube (Iris.cube.Cube):
+                Cube containing the input data to which the recursive filter
+                will be applied.
+            alpha (Float):
+                The constant used to weight the recursive filter in that
+                direction: Defined such that 0.0 <= alpha <= 1.0
+            alphas_cube (Iris.cube.Cube or None):
+                Cube containing array of alpha values that will be used
+                when applying the recursive filter in a specific direction.
+
+        Raises:
+            ValueError: If dimension of alphas array is less than dimension
+                        of data array
+            ValueError: If dimension of alphas array is greater than dimension
+                        of data array
+
+        Returns:
+            alphas_cube (Iris.cube.Cube):
+                Cube containing a padded array of alpha values
+                for the specified direction.
+        """
+
```

Comment:
Needs some kind of error check for the case when neither the float (alpha) or the cube (alphas_cube) is provided. This will need a unit test as well.